### PR TITLE
adapter: Buffer updates to metrics system tables

### PR DESCRIPTION
**Pull request by jkosh44**
Previously whenever we needed to update a system table containing metrics, we would immediately trigger a potentially expensive group commit whenever a metric was updated. Metrics are not critical enough that we needed to update the system table immediately.

This commit update the append logic so the table updates for metrics are buffered but don't trigger a group commit. Instead, they will piggyback off the next group commit. This will hopefully allow multiple metrics updates to be coalesced into a single group commit.

This commit also restarts the table advancement timer whenever a system table is updated, which was left off of a previous commit. In the previous commit only user writes would reset the timer.

### Motivation
This PR adds a feature that has not yet been specified.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/adapter/src/coord/appends.rs [hunk 1]
```diff
@@ -375,27 +375,42 @@ impl Coordinator {
...
         // This can cause future queries to block, but will not affect correctness. Since this
         // rate of DDL is unlikely, we allow DDL to explicitly trigger group commit.
-        self.pending_writes
-            .push(PendingWriteTxn::System { updates, source });
+        self.pending_writes.push(PendingWriteTxn::System {
+            updates,
+            source: BuiltinTableUpdateSource::DDL,
+        });
         self.group_commit_initiate(None).await;
+        self.advance_timelines_interval.reset();
```
Comment (jkosh44) [inline src/adapter/src/coord/appends.rs:413]: I accidentally left this out of the previous one which is pretty crucial.

**Comment by danhhz** [comment]
cool! and the 1 sec ticker is implemented by triggering group commit, which will then flush these out if nothing else has? seems great!

**Comment by jkosh44** [comment]
> cool! and the 1 sec ticker is implemented by triggering group commit, which will then flush these out if nothing else has? seems great!

Yes exactly.

File: src/adapter/src/coord/appends.rs (hunk 1 above)
Comment (jkosh44) [reply src/adapter/src/coord/appends.rs:413]: I'm going to extract this out into it's own PR because I just left it off the previous one.

**Comment by danhhz** [comment]
spun this up on my staging env (82803c07) to confirm whether we get the expected reduction in CaA calls. only a couple minutes of data so far (so grain of salt caveat), but looking like ~1.5 CaS/shard/s, which isn't quite the ~1 we expect, but far better than the ~2.5 we're seeing before these two PRs

**Comment by danhhz** [comment]
correction! I was looking at cmd/s but CaA/s is ~52 for 48 shards, which I'd certainly call close enough \o/

File: src/adapter/src/coord/message_handler.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ use mz_storage_client::controller::CollectionMetadata;
 
 use crate::client::ConnectionId;
 use crate::command::{Command, ExecuteResponse};
-use crate::coord::appends::{BuiltinTableUpdateSource, Deferred};
```
Comment (mjibson) [inline src/adapter/src/coord/message_handler.rs:32]: Can the whole `BuiltinTableUpdateSource` enum go away now?

**Review by mjibson — APPROVED** [review]

File: src/adapter/src/coord/message_handler.rs (hunk 2 above)
Comment (jkosh44) [reply src/adapter/src/coord/message_handler.rs:32]: We still use it within group commit to determine if we should wait for the append to complete in an asynchronous task or in the main coord thread.

**Comment by pH14** [comment]
This is so exciting! Thank you so much @jkosh44 🙇

**Pull request merged**